### ISSUE TITLE
Configuration : behavior change between 21.6 & 21.2 for unknown parameter

### ISSUE THREAD

**Issue by ylorph** [EventStore/EventStore#3028]
**Describe the bug**
an unknown options in 21.2 would let the server start .
in 21.6 an unknown option prevents the server from starting
reported here : https://discuss.eventstore.com/t/fatal-error-on-unknown-option/3383/2

**To Reproduce**
1. set env variable to  EVENTSTORE_HTTP_SERVICE_PROTOCOL=http
2. start a 21.2.0 server 
3. start a 21.6.0 server

**Expected behavior**
21.6.0  starts with a warning about the unknwon settings

**Actual behavior**
21.6.0 does not start 
`Error while parsing options: The option HttpServiceProtocol is not a known option. (Parameter 'HttpServiceProtocol')`

**Config/Logs/Screenshots**

**EventStore details**
 - EventStore server version:

 - Operating system: WIN10

 - EventStore client version (if applicable):

**Additional context**
21.2 & 21.6
when passing an unknwon  parameter through & config file  ´--config config.yml´ it prevents the server from starting as well 

this is inconsistent behavior on 21.2

**Comment by stellirin** [comment]
I spent some hours today when I attempted to upgrade from 21.2:

```
Error while parsing options: The option 3ServicePortTcpInt is not a known option. (Parameter '3ServicePortTcpInt')
```

We run EventStore on Kubernetes where its standard behaviour is to inject to the pod many environment variables about services on the cluster. We have services named `eventstore-1`, `eventstore-2` etc. so the pod gets a range of environment variables such as:

```
EVENTSTORE_3_SERVICE_HOST=10.0.0.123
EVENTSTORE_3_SERVICE_PORT=1112
EVENTSTORE_3_SERVICE_PORT_TCP_INT=1112
EVENTSTORE_3_SERVICE_PORT_TCP_EXT=1113
EVENTSTORE_3_SERVICE_PORT_TCP_HTTP=2113
EVENTSTORE_3_PORT_1112_TCP=tcp://10.0.0.123:1112
EVENTSTORE_3_PORT_1113_TCP=tcp://10.0.0.123:1113
EVENTSTORE_3_PORT_2113_TCP=tcp://10.0.0.123:2113
EVENTSTORE_3_PORT=tcp://10.0.0.123:1112
EVENTSTORE_3_PORT_1112_TCP_ADDR=10.0.0.123
EVENTSTORE_3_PORT_1113_TCP_ADDR=10.0.0.123
EVENTSTORE_3_PORT_2113_TCP_ADDR=10.0.0.123
EVENTSTORE_3_PORT_1112_TCP_PORT=1112
EVENTSTORE_3_PORT_1112_TCP_PROTO=tcp
EVENTSTORE_3_PORT_1113_TCP_PORT=1113
EVENTSTORE_3_PORT_1113_TCP_PROTO=tcp
EVENTSTORE_3_PORT_2113_TCP_PORT=2113
EVENTSTORE_3_PORT_2113_TCP_PROTO=tcp
```

So `3ServicePortTcpInt` comes from `EVENTSTORE_3_SERVICE_PORT_TCP_INT`, one of those standard environment variables.

It looks like the only way forward for us is to completely rename every Kubernetes resource related to eventstore, which effectively means to migrate all oru data to a new cluster, and all of the consequences this will have.

**Comment by lstyles** [comment]
In the meantime you can add `enableServiceLinks: false` to the pod spec in the template to stop kubernetes from creating those env vars if you don't want/need them.

https://kubernetes.io/docs/concepts/services-networking/connect-applications-service/

**Comment by hayley-jean** [comment]
We've added a new option, `--allow-unknown-options` which should resolve this. Enabling this option will cause EventStoreDB to simply log out any unrecognised options rather than stopping the server. Check https://github.com/EventStore/EventStore/pull/3652 for more info

It's currently available in version 22.10.0, and will be available in 22.10.9 when it's released
